data-raw/simulating-DAG.R

```diff
@@ -28,18 +85,50 @@ simulated_data <- simulateSEM(graph, N = 2500) %>%
     setNames(paste0("metabolite_", 1:length(.)))
 
 devtools::use_data(simulated_data, overwrite = TRUE)
+>>>>>>> 6d0f48698be38b8c1d900db5c10e53eee6b09a3d
 
 
-X_DAG<-est.pcor.skel.DAG.adj(dat=x,alpha_val=0.05)
-plot(X_DAG$skel_est)
+    # calculate censoring variable
+    cens = matrix(0,n,)
+    for(i in 1:n){
+        if (fup_time[i] < p10){cens[i]<-1}
+        else {fup_time[i] <-p10}
+    }
+    cens<-as.numeric(cens)
+    #print(mean(fup_time))
+    print(max(fup_time))
+    #print(min(fup_time))
+    print(sum(cens))
+    simSurv<-bind_cols(data.frame(fup_time,cens))
+}
+#Simulate survival times that depend on specific variables in the DAG-based data in x
+SurvTime<-c()
+DAG_SURV_EXP<-data.frame()
+SURV<-c()
+## generate a survival object where survtime depends on influential variables (IV) with specific betas
+SurvTime<-data.frame(survsim.cw(object=x,IV1="c" ,IV2= "k",IV3= "p",beta1=0.5 ,beta2=0.5 ,beta3= -0.5))
+## merge with exposure & network data
+x_SurvTime<-bind_cols(x,SurvTime)
+### check: are specified network variables associated with survival-time
+### fit<-coxph(Surv(fup_time,cens)~c+k+p,data =x_SurvTime )
+### summary(fit)
 
 
-X_GLas<-glasso(as.matrix(cov(x)), rho=.2)
-adjM_GLas<-X_GLas$wi
-colnames(adjM_GLas)<-colnames(x)
-rownames(adjM_GLas)<-colnames(x)
-GLas_net<-graph_from_adjacency_matrix(adjM_GLas, mode = c("undirected"), weighted = TRUE, diag = FALSE,
-                                     add.colnames = NULL, add.rownames = NA)
-tkplot(GLas_net, vertex.size=20, vertex.color="white")
+#Option 1: add non-informative variables
+#x_SurvTime1<-x_SurvTime%>%mutate(NoEffect1=rnorm(20000,mean=0,sd=1),ID=c(1:20000))
 
+##Option 2: creating a case-cohort
```

Comment:
If these comments are being used, let's just remove them. Keep only the code that will be used.